torchvision/transforms/v2/functional/_utils.py

```diff
@@ -1,8 +1,110 @@
+    def wrapper(inpt, *args, **kwargs):
+        input_type = type(inpt)
+
+        if needs_args_kwargs_handling:
+            # Convert args to kwargs to simplify further processing
+            kwargs.update(dict(zip(dispatcher_params, args)))
+            args = ()
+
+            # drop parameters that are not relevant for the kernel, but have a default value
+            # in the dispatcher
+            for kwarg in kwargs.keys() - kernel_params:
+                del kwargs[kwarg]
+
+            # add parameters that are passed implicitly to the dispatcher as metadata,
+            # but have to be explicit for the kernel
+            for kwarg in input_type.__annotations__.keys() & kernel_params:
+                kwargs[kwarg] = getattr(inpt, kwarg)
+
+        output = kernel(inpt.as_subclass(torch.Tensor), *args, **kwargs)
+
+        if isinstance(inpt, datapoints.BoundingBox) and isinstance(output, tuple):
+            output, spatial_size = output
+            metadata = dict(spatial_size=spatial_size)
+        else:
+            metadata = dict()
+
+        return input_type.wrap_like(inpt, output, **metadata)
+
+    return wrapper
+
+
+def _register_kernel_internal(dispatcher, datapoint_cls, *, wrap_kernel=True):
+    registry = _KERNEL_REGISTRY.setdefault(dispatcher, {})
+    if datapoint_cls in registry:
+        raise TypeError(
+            f"Dispatcher '{dispatcher.__name__}' already has a kernel registered for type '{datapoint_cls.__name__}'."
+        )
+
+    def decorator(kernel):
+        registry[datapoint_cls] = _kernel_wrapper_internal(dispatcher, kernel) if wrap_kernel else kernel
+        return kernel
+
+    return decorator
+
+
+def register_kernel(dispatcher, datapoint_cls):
+    return _register_kernel_internal(dispatcher, datapoint_cls, wrap_kernel=False)
+
+
+def _noop(inpt, *args, **kwargs):
+    return inpt
+
+
+def _register_explicit_noops(dispatcher, *datapoints_clss):
+    """
+    Although this looks redundant with the no-op behavior of _get_kernel, this explicit registration prevents users
+    from registering kernels for builtin datapoints on builtin dispatchers that rely on the no-op behavior.
+
+    For example, without explicit no-op registration the following would be valid user code:
+
+    .. code::
+        from torchvision.transforms.v2 import functional as F
+
+        @F.register_kernel(F.adjust_brightness, datapoints.BoundingBox)
+        def lol(...):
+            ...
+    """
+    for datapoint_cls in datapoints_clss:
+        register_kernel(dispatcher, datapoint_cls)(_noop)
+
+
+def _get_kernel(dispatcher, datapoint_cls):
+    registry = _KERNEL_REGISTRY.get(dispatcher)
+    if not registry:
+        raise ValueError(f"No kernel registered for dispatcher '{dispatcher.__name__}'.")
+
+    if datapoint_cls in registry:
+        return registry[datapoint_cls]
+
+    for registered_cls, kernel in registry.items():
+        if issubclass(datapoint_cls, registered_cls):
+            return kernel
```

Comment:
Correct. We currently don't have anything like that, but

1. We were talking about a `class DetectionMask(Mask)` and `SegmentationMask(Mask)`. Unless we want to register them separately instead of having one kernel for `Mask`'s, we need to keep this loop.
2. Users can do something like `class MyImage(Image)`